### PR TITLE
Record total number of ATP/HTTP/File bytes downloaded during session

### PR DESCRIPTION
**Pull request by utkarshgautamnyu**
Records total bytes in the stats and sends the data over to metaverse API.

Testing Notes:
Run Interface
Check Metabase ( http://metabase.highfidelity.com/)
Get session ID for the particular user ID
Check User Activities for the obtained session ID
In user Activities check the row with Action = stats
Sample line: "bytesDownloaded": {"atp": 0, "file": 27567737, "http": 106976449, "total": 134544186}

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1548/HighFidelity-Beta-PR10939-8e5bccb9765bc303bf275d234447865078038314.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1548/HighFidelity-Beta-PR10939-8e5bccb9765bc303bf275d234447865078038314.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1550/HighFidelity-Beta-PR10939-91722f3de3400b38b895524d43dc817ccff36d24.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1550/HighFidelity-Beta-PR10939-91722f3de3400b38b895524d43dc817ccff36d24.exe

**Comment by Atlante45** [comment]
@utkarshgautamnyu It looks like your editor is configured with tabs.

File: libraries/networking/src/AssetResourceRequest.cpp [hunk 1]
```diff
@@ -201,6 +204,9 @@ void AssetResourceRequest::onDownloadProgress(qint64 bytesReceived, qint64 bytes
 
         _lastProgressDebug = now;
     }
+	
+	
+
```
Comment (birarda) [inline libraries/networking/src/AssetResourceRequest.cpp]: why the addition of 3 empty lines here?

File: libraries/networking/src/AssetResourceRequest.cpp [hunk 2]
```diff
@@ -187,6 +187,9 @@ void AssetResourceRequest::onDownloadProgress(qint64 bytesReceived, qint64 bytes
     emit progress(bytesReceived, bytesTotal);
 
     auto now = p_high_resolution_clock::now();
+	
+    //Recording ATP bytes downloaded in stats
```
Comment (birarda) [inline libraries/networking/src/AssetResourceRequest.cpp]: space after //

File: libraries/networking/src/FileResourceRequest.cpp [hunk 3]
```diff
@@ -68,6 +69,8 @@ void FileResourceRequest::doSend() {
 
     if (_result == ResourceRequest::Success) {
         statTracker->incrementStat(STAT_FILE_REQUEST_SUCCESS);
+	    //Recording FILE bytes downloaded in stats
```
Comment (birarda) [inline libraries/networking/src/FileResourceRequest.cpp]: space after //

File: libraries/networking/src/FileResourceRequest.cpp [hunk 4]
```diff
@@ -53,6 +53,7 @@ void FileResourceRequest::doSend() {
                     }
 
                     _result = ResourceRequest::Success;
+				    fileSize=file.size();
```
Comment (birarda) [inline libraries/networking/src/FileResourceRequest.cpp]: spaces around `=`

Comment (birarda) [inline libraries/networking/src/FileResourceRequest.cpp]: Indentation on this line looks off, should be over in line with `_result`

File: libraries/networking/src/FileResourceRequest.cpp [hunk 5]
```diff
@@ -68,6 +69,8 @@ void FileResourceRequest::doSend() {
 
     if (_result == ResourceRequest::Success) {
         statTracker->incrementStat(STAT_FILE_REQUEST_SUCCESS);
+	    //Recording FILE bytes downloaded in stats
+	    statTracker->updateStat(STAT_FILE_RESOURCE_TOTAL_BYTES,fileSize);
```
Comment (birarda) [inline libraries/networking/src/FileResourceRequest.cpp]: indentation of these two lines is off

File: libraries/networking/src/HTTPResourceRequest.cpp [hunk 6]
```diff
@@ -201,6 +201,11 @@ void HTTPResourceRequest::onDownloadProgress(qint64 bytesReceived, qint64 bytesT
     _sendTimer->start();
 
     emit progress(bytesReceived, bytesTotal);
+	
+    //Recording HTTP bytes downloaded in stats
```
Comment (birarda) [inline libraries/networking/src/HTTPResourceRequest.cpp]: space after //

File: interface/src/Application.cpp [hunk 7]
```diff
@@ -1466,6 +1466,13 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer, bo
         properties["atp_mapping_requests"] = atpMappingRequests;
 
         properties["throttled"] = _displayPlugin ? _displayPlugin->isThrottled() : false;
+        QJsonObject bytesDownloaded;
```
Comment (birarda) [inline interface/src/Application.cpp]: add a space above this line

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1619/HighFidelity-Beta-PR10939-6b969bc39d784cb142f7ddb2ebbda9df1f12c562.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1619/HighFidelity-Beta-PR10939-6b969bc39d784cb142f7ddb2ebbda9df1f12c562.dmg

**Comment by CainFoool** [comment]
Testing now.

**Comment by CainFoool** [comment]
# Test Passed
Test | Result
-------- | --------
Run Interface | Pass
Check Metabase ( http://metabase.highfidelity.com/) | Pass
Get session ID for the particular user ID | Pass
Check User Activities for the obtained session ID | Pass
In user Activities check the row with Action = stats | Pass
Sample line: "bytesDownloaded": {"atp": 0, "file": 27567737, "http": 106976449, "total": 134544186} | Pass

(with help off @utkarshgautamnyu).

All good!

Let me know if this needs further testing.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Pull request merged**